### PR TITLE
[wasm][debugger] Add tests for floating common types

### PR DESCRIPTION
**Pull request by lewing**

### CODE REVIEW

**Comment by lewing** [comment]
Tests for https://github.com/mono/mono/pull/19710

File: sdks/wasm/DebuggerTestSuite/Tests.cs [hunk 1]
```diff
@@ -194,6 +194,36 @@ public class SourceList : DebuggerTestBase {
...
+				use_cfo: false,
+				test_fn: (locals) => {
+						CheckNumber (locals, "dPI", Math.PI);
+						CheckNumber (locals, "fPI", (float)Math.PI);
+						CheckNumber (locals, "iMax", int.MaxValue);
+						CheckNumber (locals, "iMin", int.MinValue);
+						CheckNumber (locals, "uiMax", uint.MaxValue);
+						CheckNumber (locals, "uiMin", uint.MinValue);
+
+						CheckNumber (locals, "l", uint.MaxValue * (long)2);
```
Comment (baulig) [inline sdks/wasm/DebuggerTestSuite/Tests.cs:211]: I think you are incorrect about this, I just tested both `long` and `ulong` with your patch and it seems to be working fine:

Screen Shot 2020-05-05 at 5 58 35 PM

File: sdks/wasm/DebuggerTestSuite/Tests.cs [hunk 2]
```diff
@@ -194,6 +194,36 @@ public class SourceList : DebuggerTestBase {
...
+						//CheckNumber (locals, "lMax", long.MaxValue); // cannot be represented as double
```
Comment (baulig) [inline sdks/wasm/DebuggerTestSuite/Tests.cs:212]:
```suggestion
						CheckNumber (locals, "lMax", long.MaxValue);
```

File: sdks/wasm/DebuggerTestSuite/Tests.cs [hunk 3]
```diff
@@ -194,6 +194,36 @@ public class SourceList : DebuggerTestBase {
...
+						//CheckNumber (locals, "lMin", long.MinValue); // cannot be represented as double
```
Comment (baulig) [inline sdks/wasm/DebuggerTestSuite/Tests.cs:213]:
```suggestion
						CheckNumber (locals, "lMin", long.MinValue);
						CheckNumber (locals, "ulMax", ulong.MaxValue);
```

File: sdks/wasm/debugger-test2.cs [hunk 4]
```diff
@@ -15,4 +15,30 @@ public class Misc { //Only append content to this class as the test suite depend
...
+		double dPI = System.Math.PI;
+		float  fPI = (float)System.Math.PI;
+
+		int iMax = int.MaxValue;
+		int iMin = int.MinValue;
+		uint uiMax = uint.MaxValue;
+		uint uiMin = uint.MinValue;
+
+		long l = uiMax * (long)2;
+		long lMax = long.MaxValue; // cannot be represented as double
```
Comment (baulig) [inline sdks/wasm/debugger-test2.cs:28]:
```suggestion
		long lMax = long.MaxValue;
```

File: sdks/wasm/debugger-test2.cs [hunk 5]
```diff
@@ -15,4 +15,30 @@ public class Misc { //Only append content to this class as the test suite depend
...
+		long lMin = long.MinValue; // cannot be represented as double
```
Comment (baulig) [inline sdks/wasm/debugger-test2.cs:29]:
```suggestion
		long lMin = long.MinValue;
		ulong ulMax = ulong.MaxValue;
```

**Comment by baulig** [comment]
I believe you are incorrect about `long` and `ulong`, they seem to be working fine - see my screenshot.

File: sdks/wasm/DebuggerTestSuite/Tests.cs (hunk 1 above)
Comment (baulig) [reply sdks/wasm/DebuggerTestSuite/Tests.cs:211]: Sorry, should have looked at the actual values - they are incorrect.

**Comment by lewing** [comment]
@monojenkins backport 3.2-wasm

**Pull request merged**